src/orchestrator-kubernetes/src/lib.rs

```diff
@@ -1565,19 +1594,33 @@ impl OrchestratorWorker {
         ret.await
     }
 
-    async fn ensure_service(&self, desc: &ServiceDescription) -> Result<(), K8sError> {
+    async fn ensure_service(&self, mut desc: ServiceDescription) -> Result<(), K8sError> {
+        // We inject our own owner references into the Kubernetes objects
+        // created for the service so that if the pod hosting this orchestrator
+        // gets deleted, so do all services spawned by this orchestrator.
```

Comment:
nit: I think it should be if the statefulset (of a particular generation) gets deleted, we want to clear up associated resources, rather than if the pod gets deleted. afaict this is what the code does currently but isn't 1:1 with this comment

Reply:
Ah good point. Updated the comment.